app/scripts/directives/mobileClientBuildsList.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+(function() {
+  angular.module('openshiftConsole').component('mobileClientBuildsList', {
+    controller: [
+      '$timeout',
+      '$anchorScroll',
+      '$location',
+      'APIService',
+      'DataService',
+      MobileClientBuildsList
+    ],
+    bindings: {
+      mobileClient: '<'
+    },
+    templateUrl: 'views/mobile-client-builds-list.html'
+  });
+
+  function MobileClientBuildsList(
+    $timeout,
+    $anchorScroll,
+    $location,
+    APIService,
+    DataService
+  ) {
+    var ctrl = this;
+    var buildConfigsVersion = APIService.getPreferredVersion('buildconfigs');
+    var watches = [];
+    var anchor;
+
+    ctrl.$onChanges = function(changes) {
+      var mobileClientChanged = _.get(changes, "mobileClient.currentValue", false);
+      if (mobileClientChanged && !ctrl.context) {
+        ctrl.context = {namespace: _.get(ctrl, 'mobileClient.metadata.namespace')};
+      }
+
+      if (mobileClientChanged && !ctrl.buildsWatchSet) {
+        ctrl.buildsWatchSet = true;
+        watches.push(DataService.watch(buildConfigsVersion, ctrl.context, function(buildConfigs) {    
+          ctrl.mobileBuildConfigs = _.filter(buildConfigs.by('metadata.name'), function(buildConfig) {
+            return _.get(buildConfig, 'metadata.labels.mobile-client-build') === 'true' && _.get(buildConfig, 'metadata.labels.mobile-client-id') === _.get(ctrl, 'mobileClient.metadata.name');
+          });
+          ctrl.loaded = true;
+          ctrl.goToBuildConfig();
+        }));
+      }
+    };
+
+    ctrl.goToBuildConfig = function() {
+      if ($location.hash() && anchor !== $location.hash()) {
+        anchor = $location.hash();
+
+        $timeout(function() {
+          $anchorScroll.yOffset = 100;
+          $anchorScroll();
+        }, 400);
```

Comment:
Is the `400` arbitrary?

Reply:
It is arbitrary. This handles the case where the DOM is still being generated and the anchor doesn't exist yet. I am not sure this is the best way to handle this, have you any examples of handling this elsewhere?

Reply:
But you are setting `anchor` to `$location.hash()`, do you mean an html anchor? 

This is a bit more confusing in that the function is `goToBuildConfig()` but all of the contents are in a single `if`. Is there a potential that this will be clicked but nothing will happen?

Ah, hold on, this isn't a `view` callback. Perhaps just do `var goToBuildConfig` to make it clear this isn't a callback.